### PR TITLE
Keep GitHub Actions up to date with GitHub's Dependabot

### PR DESCRIPTION
**Pull request by cclauss**
Auto-generates pull requests like
* #4157

Which will fix warnings like at the bottom right of https://github.com/pyodide/pyodide/actions/runs/8306232175

* https://docs.github.com/en/code-security/dependabot/working-with-dependabot/keeping-your-actions-up-to-date-with-dependabot
* https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#package-ecosystem

### Checklists

- [ ] Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry
- [ ] Add / update tests
- [ ] Add new / update outdated documentation

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+version: 2
+updates:
+  - package-ecosystem: github-actions
+    directory: /
+    groups:
+      github-actions:
+        patterns:
+          - "*" # Group all Actions updates into a single larger pull request
+    schedule:
+      interval: weekly
```
Comment (ryanking13) [inline .github/dependabot.yml:13]: Maybe let's set the interval to monthly. Weekly PRs that update GHA are sometimes cumbersome, and for that reason, we've moved to pre-commit-config updates from weekly to quarterly as well.

**Review by ryanking13 — APPROVED** [review]
Thanks @cclauss! This is going to be very convenient.

File: .github/dependabot.yml (hunk 1 above)
Comment (ryanking13) [reply .github/dependabot.yml:13]: I see. That makes sense. Thanks @cclauss!

**Comment by cclauss** [comment]
Generated:
* #4630

**Pull request merged**